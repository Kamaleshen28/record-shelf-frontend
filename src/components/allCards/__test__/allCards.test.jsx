import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import axios from 'axios';
import { useNavigate } from 'react-router-dom';
import AllCards from '../allCards';

jest.mock('react-router-dom');

const mockSongData = {
    data:[ 
        {
            'id': '0e2ddccf-f494-4c41-a499-90f8267f491a',
            'name': 'Beautiful Mistakes (feat. Megan Thee Stallion)',
            'genre': {
                'id': '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
                'name': 'Pop'
            },
            'artist': {
                'id': '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
                'name': 'Maroon 5'
            },
            'imageUrl': 'https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631',
            'publishedAt': '2021-06-11T00:00:00'
        },
        {
            'id': '45e1d753-2986-43cb-9b9d-30c370056319',
            'name': 'This Love',
            'genre': {
                'id': '128aa7f8-c943-48ce-b352-7edd26fa4c6e',
                'name': 'Pop'
            },
            'artist': {
                'id': '496b0a85-2bfa-45bc-8d0f-57fe0ce55708',
                'name': 'Maroon 5'
            },
            'imageUrl': 'https://i.scdn.co/image/ab67616d0000b27317b3850d758fff5a2301e537',
            'publishedAt': '2002-06-25T00:00:00'
        }
    ]
  
};


describe('allCards', () => {
    it('should render allCards correctly', async () => {
        jest.spyOn(axios, 'get').mockResolvedValue({data:mockSongData});
        const {asFragment} = render(<AllCards />);
        await waitFor(() => {
            expect(screen.getAllByText('Pop')).toBeTruthy();
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call naviagte with /genre when genre-icon is clicked', async () => {
        jest.spyOn(axios, 'get').mockResolvedValue({data:mockSongData});
        const navigateFn = jest.fn();
        useNavigate.mockReturnValue(navigateFn);
        render(<AllCards />);
        await waitFor(() => {
            expect(screen.getAllByText('Pop')).toBeTruthy();
        });
        const genreIcon = screen.getByTestId('genre-icon');
        fireEvent.click(genreIcon);
        expect(navigateFn).toBeCalledWith('/genre');
    });
});



