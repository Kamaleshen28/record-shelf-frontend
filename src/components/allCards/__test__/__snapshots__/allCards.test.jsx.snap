// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allCards should render allCards correctly 1`] = `
<DocumentFragment>
  <div
    class="body-dummy"
  >
    <div
      class="body-all-songs"
    >
      <div>
        <div
          class="body-header"
        >
          <p
            class="all-songs-text"
          >
            all songs
          </p>
          <img
            alt="Not Found"
            class="genre-icon"
            data-testid="genre-icon"
            src="icon-genre.svg"
          />
        </div>
        <div
          class="cards-container"
        >
          <div
            class="card light-card"
          >
            <div
              class="card-image-container"
            >
              <img
                alt="Not Found"
                class="card-image"
                src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
              />
            </div>
            <div
              class="details-container"
            >
              <div
                class="text-container"
              >
                <a
                  class="song-name-text"
                >
                  Beautiful Mistakes (feat. Megan Thee Stallion)
                </a>
                <a
                  class="song-genre-text"
                >
                  Pop
                </a>
              </div>
              <div
                class="heart-image-container"
              >
                <img
                  alt="Not Found"
                  class="heart-image"
                  data-testid="heart-image"
                  src="heart-gray.svg"
                />
                <a
                  class="count-text"
                  data-testid="like-count"
                />
              </div>
            </div>
          </div>
          <div
            class="card dark-card"
          >
            <div
              class="card-image-container"
            >
              <img
                alt="Not Found"
                class="card-image"
                src="https://i.scdn.co/image/ab67616d0000b27317b3850d758fff5a2301e537"
              />
            </div>
            <div
              class="details-container"
            >
              <div
                class="text-container"
              >
                <a
                  class="song-name-text"
                >
                  This Love
                </a>
                <a
                  class="song-genre-text"
                >
                  Pop
                </a>
              </div>
              <div
                class="heart-image-container"
              >
                <img
                  alt="Not Found"
                  class="heart-image"
                  data-testid="heart-image"
                  src="heart-gray.svg"
                />
                <a
                  class="count-text"
                  data-testid="like-count"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
